chart/templates/processor-config.yaml

```diff
@@ -5,4 +5,28 @@ metadata:
   labels:
     {{- include "kapel.labels" . | nindent 4 }}
 data:
+{{ if kindIs "string" .Values.processor.config }}
+# Deprecated: Specify the whole ConfigMap as a multiline string
 {{ .Values.processor.config | indent 2 }}
+{{ else }}
+  SITE_NAME: {{ .Values.processor.config.siteName }}
+  SUBMIT_HOST : {{ .Values.processor.config.submitHost }}
+  NAMESPACE: {{ .Values.processor.config.namespace }}
+  VO_NAME: {{ .Values.processor.config.voName }}
+  INFRASTRUCTURE_TYPE: {{ .Values.processor.config.infrastructureType }}
+  OUTPUT_PATH: {{ .Values.processor.config.outputPath }}
+
+  PROMETHEUS_SERVER: {{ .Values.processor.prometheus.server }}
+
+  PUBLISHING_MODE: {{ .Values.processor.query.mode }}
+  QUERY_TIMEOUT: {{ .Values.processor.query.timeout }}
+  {{ if eq .Values.processor.query.mode "gap"}}
+  QUERY_START: {{ .Values.processor.query.startDate }}
+  QUERY_END: {{ .Values.processor.query.endDate }}
+  {{ end }}
+  {{ if eq .Values.outputFormat "ssmsend"}}
+  BENCHMARK_VALUE: {{ .Values.ssmsend.config.benchmarkValue  | quote }}
+  PROCESSORS: {{ .Values.ssmsend.config.processors | quote  }}
+  NODECOUNT: {{ .Values.ssmsend.config.nodeCount | quote }}
+  {{ end }}
+{{ end }}
```

Comment:
How obvious are the failure modes for misconfigurations?

Reply:
There will be python messages or exceptions. The configuration validation is done here: https://github.com/rptaylor/kapel/blob/master/python/KAPELConfig.py